cl/lib/zoho.py

```diff
@@ -0,0 +1,100 @@
+from django.conf import settings
+from django.core.cache import cache
+from zohocrmsdk.src.com.zoho.api.authenticator import OAuthToken
+from zohocrmsdk.src.com.zoho.crm.api import (
+    HeaderMap,
+    Initializer,
+    ParameterMap,
+)
+from zohocrmsdk.src.com.zoho.crm.api.record import (
+    APIException,
+    RecordOperations,
+    ResponseWrapper,
+    SearchRecordsParam,
+)
+
+
+def get_zoho_cache_key() -> str:
+    return "zoho_token"
+
+
+class SearchableModule:
+    module_name: str = ""
+
+    @staticmethod
+    def initialize():
+        refresh_token = cache.get(f"{get_zoho_cache_key()}:refresh")
+        if not refresh_token:
+            raise Exception(
+                f"Cache miss: no value found for key {get_zoho_cache_key()}:refresh"
+            )
+        token = OAuthToken(
+            client_id=settings.ZOHO_CLIENT_ID,
+            client_secret=settings.ZOHO_CLIENT_SECRET,
+            refresh_token=refresh_token,
+        )
+        Initializer.initialize(
+            environment=settings.ZOHO_ENV,
+            token=token,
+            store=settings.ZOHO_STORE,
+            resource_path=settings.ZOHO_RESOURCE_PATH,
+        )
+
+    def get_record_by_cl_id_or_email(
+        self, cl_ids: list[int], email: list[str]
+    ):
+        record_operations = RecordOperations(self.module_name)
+        param_instance = ParameterMap()
+
+        ids_str = ",".join([str(i) for i in cl_ids])
+        emails_str = ",".join(email)
+
+        criteria = []
+        if emails_str:
+            criteria.append(f"(Email:in:{emails_str})")
+        if ids_str:
+            criteria.append(f"(CourtListener_ID:in:{ids_str})")
+
+        criteria_str = " or ".join(criteria)
+        param_instance.add(SearchRecordsParam.criteria, f"({criteria_str})")
+
+        header_instance = HeaderMap()
+        response = record_operations.search_records(
+            param_instance, header_instance
+        )
+
+        if response is None:
+            raise Exception("Received no response from the API.")
+
+        status_code = response.get_status_code()
+        if status_code in [204, 304]:
+            msg = "No Content" if status_code == 204 else "Not Modified"
+            raise Exception(f"Search returned no records ({msg}).")
```

Comment:
Tweak this to make the error clear it's from zoho? 


```suggestion
            raise Exception(f"Zoho query returned no records ({msg}).")
```